tap_zendesk_chat/streams.py

```diff
@@ -21,81 +16,71 @@ class Stream:
-            self.write_page(page)
+    def sync(self, ctx, schema: Dict, stream_metadata: Dict, transformer: Transformer):
+        response = ctx.client.request(self.tap_stream_id)
+        page = [transformer.transform(rec, schema, metadata=stream_metadata) for rec in response]
+        self.write_page(page)
 
 
 class Agents(Stream):
-    def sync(self, ctx):
+    def sync(self, ctx, schema: Dict, stream_metadata: Dict, transformer: Transformer):
         since_id_offset = [self.tap_stream_id, "offset", "id"]
         since_id = ctx.bookmark(since_id_offset) or 0
-        schema = ctx.catalog.get_stream(self.tap_stream_id).schema.to_dict()
-        m_data = metadata.to_map(ctx.catalog.get_stream(self.tap_stream_id).metadata)
-
-        with Transformer() as transformer:
-            while True:
-                params = {
-                    "since_id": since_id,
-                    "limit": ctx.config.get("agents_page_limit", 100),
-                }
-                page = ctx.client.request(self.tap_stream_id, params)
-                if not page:
-                    break
-                page = [transformer.transform(rec, schema, metadata=m_data) for rec in page]
-                self.write_page(page)
-                since_id = page[-1]["id"] + 1
-                ctx.set_bookmark(since_id_offset, since_id)
-                ctx.write_state()
+        while True:
+            params = {
+                "since_id": since_id,
+                "limit": ctx.config.get("agents_page_limit", 100),
+            }
+            page = ctx.client.request(self.tap_stream_id, params)
+            if not page:
+                break
+            self.write_page([transformer.transform(rec, schema, metadata=stream_metadata) for rec in page])
+            since_id = page[-1]["id"] + 1
+            ctx.set_bookmark(since_id_offset, since_id)
+            ctx.write_state()
         ctx.set_bookmark(since_id_offset, None)
         ctx.write_state()
 
 
 class Chats(Stream):
-    replication_key = {'timestamp', 'end_timestamp'}
-    def _bulk_chats(self, ctx, chat_ids):
+    replication_key = {"timestamp", "end_timestamp"}
+    forced_replication_method = "INCREMENTAL"
+
+    def _bulk_chats(self, ctx, chat_ids: List):
         if not chat_ids:
             return []
         params = {"ids": ",".join(chat_ids)}
         body = ctx.client.request(self.tap_stream_id, params=params)
         return list(body["docs"].values())
 
-    def _search(self, ctx, chat_type, ts_field,
-                start_dt: datetime, end_dt: datetime):
-        params = {
-            "q": "type:{} AND {}:[{} TO {}]"
-            .format(chat_type, ts_field, start_dt.isoformat(), end_dt.isoformat())
-        }
-        return ctx.client.request(
-            self.tap_stream_id, params=params, url_extra="/search")
+    def _search(self, ctx, chat_type, ts_field, start_dt: datetime, end_dt: datetime):
```

Comment:
there is an inconsistency with type hinting, it's missing for`ctx`, `chat_type`, `ts_field` args here and same scenario in other methods and other files as well

Reply:
i have removed this method from the code